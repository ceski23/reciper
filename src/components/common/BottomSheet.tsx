/* eslint-disable @typescript-eslint/no-use-before-define */
import styled from '@emotion/styled';
import { animated, useSpring } from '@react-spring/web';
import { useDrag } from '@use-gesture/react';
import {
  FC, useCallback, useEffect, useState,
} from 'react';

import { ReactPortal } from 'components/common/modal/ReactPortal';

import { exhaustiveCheck } from 'utils/guards';
import { color } from 'utils/styles/theme';

export type SheetState = 'open' | 'close' | 'peek';

interface BottomSheetProps {
  state: SheetState;
  onStateChange: (state: SheetState) => void;
  disableDrag?: boolean;
  gap?: number;
}

export const BottomSheet: FC<BottomSheetProps> = ({
  state, onStateChange, children, disableDrag, gap = 40,
}) => {
  const [sheetHeight, setSheetHeight] = useState(0);
  const [showBackdrop, setShowBackdrop] = useState(false);

  const measuredRef = useCallback((node: HTMLElement | null) => {
    if (node) setSheetHeight(node.getBoundingClientRect().height);
  }, []);

  const backdropStyles = useSpring({ opacity: state !== 'close' ? 1 : 0 });
  const [{ y }, api] = useSpring({
    y: 0,
    onStart(result) {
      if (result.value.y !== 0) setShowBackdrop(true);
    },
    onRest(result) {
      if (result.value.y === 0) setShowBackdrop(false);
    },
  }, []);

  const openSheet = useCallback(() => {
    api.start({ y: -sheetHeight + gap, immediate: false });
    onStateChange('open');
  }, [api, gap, onStateChange, sheetHeight]);

  const closeSheet = useCallback(() => {
    api.start({ y: 0, immediate: false });
    onStateChange('close');
  }, [api, onStateChange]);

  const peekSheet = useCallback(() => {
    api.start({ y: -sheetHeight * 0.4, immediate: false });
    onStateChange('peek');
  }, [api, onStateChange, sheetHeight]);

  useEffect(() => {
    switch (state) {
      case 'open': openSheet(); break;
      case 'close': closeSheet(); break;
      case 'peek': peekSheet(); break;

      default:
        exhaustiveCheck(state, 'Not all states checked');
        break;
    }
  }, [closeSheet, openSheet, peekSheet, state]);

  const bind = useDrag(({
    velocity: [,vy], offset: [,my], direction: [,dy], last,
  }) => {
    if (disableDrag) return;

    // If finished dragging
    if (last) {
      // If dragged upwards then open sheet
      if (dy < 0) openSheet();
      // If dragged quickly downwards or opened less than 40% of size then close sheet
      else if ((dy > 0 && vy > 0.5) || my > -sheetHeight * 0.4) closeSheet();
      // Open sheet
      else openSheet();
    } else {
      api.start({ y: my, immediate: true });
    }
  }, {
    delay: true,
    from: () => [0, y.get()],
    filterTaps: true,
    bounds: {
      top: -sheetHeight,
    },
  });

  return (
    <ReactPortal wrapperId="filtersModal">
      {showBackdrop && (
        <Backdrop style={backdropStyles} onClick={() => onStateChange('close')} />
      )}

      <Container ref={measuredRef} {...bind()} style={{ y }}>
        <Handle />
        <Content>{children}</Content>
      </Container>
    </ReactPortal>

  );
};

const Container = styled(animated.div)`
  background-color: ${color('background')};
  position: fixed;
  bottom: -100vh;
  left: 0;
  right: 0;
  z-index: 99;
  height: 100vh;
  touch-action: none;
  border-radius: 10px 10px 0 0;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  align-items: center;
  box-shadow: 0px 0px 20px ${color('shadow')};
`;

const Handle = styled.div`
  background-color: ${color('backgroundAltHover')};
  width: 40px;
  height: 4px;
  border-radius: 10px;
  margin: 20px;
`;

const Backdrop = styled(animated.div)`
  background-color: #00000075;
  position: fixed;
  inset: 0;
  width: 100%;
  height: 100%;
  cursor: pointer;
  z-index: 99;
`;

const Content = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  flex: 1;
  padding: 0 20px 20px;
`;
