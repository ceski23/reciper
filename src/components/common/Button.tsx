/** @jsxImportSource @emotion/react */
import { css, Theme } from '@emotion/react/macro';
import styled from '@emotion/styled/macro';
import { hsl, parseToHsl, transparentize } from 'polished';
import React, { FC } from 'react';
import { match } from 'ts-pattern';

export type ButtonVariant = 'normal' | 'primary';
export type ButtonSize = 'normal' | 'small';

export const buttonStyles = (
  { theme, variant, size }: { theme: Theme, variant: ButtonVariant, size: ButtonSize },
) => css`
  font-weight: bold;
  font-family: Raleway, sans-serif;
  border: none;
  transition: background-color 0.3s, color 0.3s;
  outline: none;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: auto;
  cursor: pointer;
  box-shadow: 0px 3px 2px #0000001c;
  
  &:disabled {
    opacity: 0.7;
    cursor: not-allowed;
  }

  ${match(variant)
    .with('normal', () => css`
      background-color: ${theme.colors.backgroundhover};
      color: ${theme.colors.textalt};

      &:disabled {
        background-color: ${theme.colors.background2};
      }

      &:not(:disabled) {
        &:hover,
        &:focus-visible {
          background-color: ${hsl({
    hue: parseToHsl(theme.colors.backgroundhover).hue,
    saturation: parseToHsl(theme.colors.backgroundhover).saturation,
    lightness: parseToHsl(theme.colors.backgroundhover).lightness - 0.1,
  })};
        }
      }
    `)
    .with('primary', () => css`
      background-color: ${theme.colors.primary};
      color: ${transparentize(0, theme.colors.textLight)};

      &:disabled {
        background-color: ${theme.colors.primaryDisabled};
        color: ${theme.colors.text};
      }

      &:not(:disabled) {
        &:hover,
        &:focus-visible {
          background-color: ${theme.colors.primaryHover};
        }
      }
    `)
    .exhaustive()}
  
  ${match(size)
    .with('normal', () => css`
      font-size: 18px;
      border-radius: 10px;
      padding: 15px 15px;
      line-height: 1;

      & > svg, & > img {
        margin-right: 15px;
        width: 20px;
        height: 20px;
      }

      & > .filler {
        margin-left: 15px;
        width: 20px;
        height: 20px;
      }
    `)
    .with('small', () => css`
      font-size: 14px;
      border-radius: 8px;
      padding: 12px 12px;
      line-height: 1;

      & > svg, & > img {
        margin-right: 15px;
        width: 15px;
        height: 15px;
      }

      & > .filler {
        margin-left: 15px;
        width: 15px;
        height: 15px;
      }
    `)
    .exhaustive()}
`;

export const StyledButton = styled.button<{
  variant: ButtonVariant,
  size: ButtonSize
}>(buttonStyles);

export type HTMLButtonProps = React.DetailedHTMLProps<
React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement
>;

interface Props {
  icon?: React.FunctionComponent<React.SVGProps<SVGSVGElement>> | string;
  variant?: ButtonVariant;
  size?: ButtonSize;
}

export const Button: FC<HTMLButtonProps & Props> = ({
  children, icon: Icon, variant = 'normal', size = 'normal', ...props
}) => (
  <StyledButton size={size} variant={variant} {...props}>
    {Icon && (typeof Icon === 'string' ? <img src={Icon} className="filler" alt="" /> : <Icon />)}
    {children}
    {Icon && <span className="filler" />}
  </StyledButton>
);
