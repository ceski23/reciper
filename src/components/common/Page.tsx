import styled from '@emotion/styled/macro';
import { motion } from 'framer-motion';
import { FC } from 'react';

import { ReactComponent as HomeIcon } from 'assets/common/home.svg';
import { ReactComponent as SearchIcon } from 'assets/common/loupe.svg';
import { ReactComponent as ReciperIcon } from 'assets/common/reciper.svg';
import { ReactComponent as RecipesIcon } from 'assets/common/recipes.svg';
import { ReactComponent as SettingsIcon } from 'assets/settings/cogwheel.svg';

import { NavigationMenu } from 'components/navigation/NavigationMenu';
import { NavigationMenuItem } from 'components/navigation/NavigationMenuItem';

import { urls } from 'routing/urls';

import { media } from 'utils/styles/mediaQueries';
import { color } from 'utils/styles/theme';

const PageBody = styled(motion.div)`
  overflow: auto;
  grid-column: 2;
  /* display: grid;
  align-content: flex-start;
  grid-template-columns: 50px 1fr 50px;

  ${media.down('medium')} {
    grid-template-columns: 1fr 90% 1fr;
    padding-bottom: 70px;
  }

  & > * {
    grid-column: 2;
    width: 100%;
  } */
`;

const PageContainer = styled.div`
  height: 100vh;
  display: grid;
  grid-template-columns: auto 1fr;

  ${media.down('medium')} {
    grid-template-columns: 0 1fr;
  }
`;

const PageSidebar = styled.aside`
  grid-column: 1;
  width: 300px;
  display: flex;
  flex-direction: column;

  @media (max-width: 1500px) {
    width: 250px;
  }
`;

const Logo = styled.div`
  width: 100%;
  height: 100px;
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 20px;
  gap: 10px;

  ${media.down('medium')} {
    display: none;
  }
`;

const AppIcon = styled(ReciperIcon)`
  height: 100%;
  width: auto;
`;

const AppName = styled.p`
  color: ${color('primary')};
  font-size: 30px;
  font-weight: 700;
`;

export const Page: FC = ({ children }) => (
  <PageContainer>
    <PageSidebar>
      <Logo>
        <AppIcon />
        <AppName>Reciper</AppName>
      </Logo>

      <NavigationMenu>
        <NavigationMenuItem title="GÅ‚Ã³wna" icon={HomeIcon} to={urls.home} />
        <NavigationMenuItem title="Szukaj" icon={SearchIcon} to={String(urls.search)} />
        <NavigationMenuItem title="Przepisy" icon={RecipesIcon} to={String(urls.recipes)} />
        <NavigationMenuItem title="Ustawienia" icon={SettingsIcon} to={String(urls.settings)} />
      </NavigationMenu>
    </PageSidebar>

    <PageBody
      initial={{ opacity: 0 }}
      animate={{ opacity: 1 }}
      exit={{ opacity: 0 }}
      transition={{ duration: 0.2 }}
    >
      {children}
    </PageBody>
  </PageContainer>
);
