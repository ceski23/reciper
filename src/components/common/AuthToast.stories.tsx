import { ComponentMeta, ComponentStory } from '@storybook/react';

import { AuthToast } from 'components/common/AuthToast';

export default {
  title: 'Components/AuthToast',
  component: AuthToast,
} as ComponentMeta<typeof AuthToast>;

const Template: ComponentStory<typeof AuthToast> = (args) => <AuthToast {...args} />;

export const Default = Template.bind({});
Default.args = {};
