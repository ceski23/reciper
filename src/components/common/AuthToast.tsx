import styled from '@emotion/styled';
import { VFC } from 'react';

import { Link } from 'components/common/Link';

import { urls } from 'routing/urls';

import { color } from 'utils/styles/theme';

export const CustomToastIcon = styled.img`
  width: 20px;
  height: 20px;
`;

const ToastContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  width: 100%;
`;

const ToastLink = styled(Link)`
  margin-left: 20px;
  background-color: ${color('backgroundHover')};
  padding: 5px 10px;
  border-radius: 10px;
  color: ${color('textalt')};
  text-decoration: none;
  font-weight: 600;
`;

export const AuthToast: VFC = () => (
  <ToastContainer>
    <span>Wymagane ponowne uwierzytelnienie</span>
    <ToastLink to={urls.settings.account()}>OK</ToastLink>
  </ToastContainer>
);
