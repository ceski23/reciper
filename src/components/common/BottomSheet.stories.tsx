import { Meta, StoryObj } from '@storybook/react';
import { useState } from 'react';

import { BottomSheet, SheetState } from 'components/common/BottomSheet';

export default {
  title: 'Components/BottomSheet',
  component: BottomSheet,
} as ComponentMeta<typeof BottomSheet>;

const Template: ComponentStory<typeof BottomSheet> = ({ state, ...args }) => {
  const [sheetState, setSheetState] = useState<SheetState>(state);

  const handleStateChange = (newState: SheetState) => {
    args.onStateChange(newState);
    setSheetState(newState);
  };

  return (
    <BottomSheet {...args} state={sheetState} onStateChange={handleStateChange} />
  );
};

export const Default = Template.bind({});
Default.args = {
  state: 'peek',
};
