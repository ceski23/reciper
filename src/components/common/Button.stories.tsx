import { ComponentMeta, ComponentStory } from '@storybook/react';

import { ReactComponent as HomeIcon } from 'assets/common/home.svg';

import { Button } from 'components/common/Button';

export default {
  title: 'Components/Button',
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Normal = Template.bind({});
Normal.args = {
  variant: 'normal',
  children: 'Button',
};

export const Primary = Template.bind({});
Primary.args = {
  variant: 'primary',
  children: 'Button',
};

export const NormalSize = Template.bind({});
NormalSize.args = {
  size: 'normal',
  children: 'Button',
};

export const SmallSize = Template.bind({});
SmallSize.args = {
  size: 'small',
  children: 'Button',
};

export const WithIcon = Template.bind({});
WithIcon.args = {
  icon: HomeIcon,
  children: 'Button',
};
