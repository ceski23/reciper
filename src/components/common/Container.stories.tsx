import { ComponentStory } from '@storybook/react';

import { FluidContainer, FullBleed } from 'components/common/Container';

const loremIpsum = `
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. In accumsan luctus interdum. Fusce in efficitur sapien. Vestibulum volutpat ut mauris accumsan cursus. Aenean sit amet malesuada purus. Duis vel maximus arcu. Nam quis est tortor. Quisque elementum elit ac augue eleifend congue. Proin consectetur eget nisi at placerat. Nulla vel consectetur dolor, non feugiat ipsum. Interdum et malesuada fames ac ante ipsum primis in faucibus. Integer aliquam libero in viverra feugiat. Nulla facilisi. Cras pharetra efficitur nisi vitae tempus. Proin pellentesque ex cursus pellentesque hendrerit. Etiam ligula sem, pretium quis augue quis, aliquam feugiat lectus.
`;

export default {
  title: 'Components/Container',
};

export const FluidStory: ComponentStory<typeof FluidContainer> = (props) => (
  <FluidContainer {...props} />
);
FluidStory.args = { children: <p>{loremIpsum}</p> };
FluidStory.storyName = 'FluidContainer';

export const FullBleedStory: ComponentStory<typeof FullBleed> = (props) => (
  <FluidContainer>
    <FullBleed {...props} />
  </FluidContainer>
);
FullBleedStory.args = { children: <p>{loremIpsum}</p> };
FullBleedStory.storyName = 'FullBleed';
