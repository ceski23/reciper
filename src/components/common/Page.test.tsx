import { renderWithProviders } from 'test/utils';

import { Page } from 'components/common/Page';

describe('Page', () => {
  it('should render correctly', () => {
    const { asFragment } = renderWithProviders(<Page />);

    expect(asFragment()).toMatchSnapshot();
  });
});
