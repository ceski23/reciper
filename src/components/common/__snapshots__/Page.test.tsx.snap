// Vitest Snapshot v1

exports[`Page > should render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  height: 100vh;
  display: grid;
  grid-template-columns: auto 1fr;
}

@media (max-width: 768px) {
  .emotion-0 {
    grid-template-columns: 0 1fr;
  }
}

.emotion-2 {
  grid-column: 1;
  width: 300px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (max-width: 1500px) {
  .emotion-2 {
    width: 250px;
  }
}

.emotion-4 {
  width: 100%;
  height: 100px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 20px;
  gap: 10px;
}

@media (max-width: 768px) {
  .emotion-4 {
    display: none;
  }
}

.emotion-6 {
  height: 100%;
  width: auto;
}

.emotion-8 {
  color: #0af;
  font-size: 30px;
  font-weight: 700;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  background-color: #f6f8f9;
  width: 100%;
  z-index: 10;
}

@media (max-width: 768px) {
  .emotion-10 {
    box-shadow: 0px 0px 20px rgba(57,66,70,0.3);
    height: 70px;
    position: fixed;
    bottom: 0;
    left: 0;
    right: 0;
  }
}

.emotion-12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-evenly;
  -ms-flex-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  justify-content: space-evenly;
  padding: 0;
  margin: 0;
  width: 100%;
  height: 100%;
  gap: 10px;
  padding: 10px;
}

@media (max-width: 768px) {
  .emotion-12 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 0;
  }
}

.emotion-14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 15px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: unset;
  font-weight: 600;
  gap: 10px;
  border-radius: 10px;
}

.emotion-14.active {
  color: #0af;
  background-color: rgba(0,170,255,0.1);
}

@media (max-width: 768px) {
  .emotion-14.active {
    background-color: transparent;
  }

  .emotion-14.active::after {
    content: '';
    display: block;
    position: absolute;
    bottom: 0;
    width: 80%;
    height: 4px;
    left: 0;
    right: 0;
    margin: auto;
    border-radius: 15px 15px 0 0;
    background-color: currentColor;
  }
}

.emotion-16 {
  width: 25px;
  fill: currentColor;
}

.emotion-17 {
  display: none;
}

@media (min-width: 576px) {
  .emotion-17 {
    display: block;
  }
}

<div
    class="emotion-0 emotion-1"
  >
    <aside
      class="emotion-2 emotion-3"
    >
      <div
        class="emotion-4 emotion-5"
      >
        <svg
          class="emotion-6 emotion-7"
          viewBox="0 0 64 64"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M55.5 52.5v10h-4v-10z"
            fill="#0092db"
          />
          <path
            d="M51.5 57.5v5H14.501c-2.76 0-5-2.24-5-5s2.24-5 5-5H51.5z"
            fill="#e6e9ed"
          />
          <path
            d="M15.5.5v52h-2c-2.76 0-5 2.24-5 5v-52c0-2.76 2.24-5 5-5z"
            fill="#656d78"
          />
          <path
            d="M15.5.5v52h40v-51l-1.001-1z"
            fill="#00aaff"
          />
          <path
            d="M54.499 53.5H14.501a1 1 0 110-2h39.998a1 1 0 110 2z"
            fill="#0092db"
          />
          <path
            d="M54.499 63.5H14.501a1 1 0 110-2h39.998a1 1 0 110 2z"
            fill="#0092db"
          />
          <path
            d="M14.501 56.502a1 1 0 100 2H51.5v-2z"
            fill="#ccd1d9"
          />
          <path
            d="M14.501 63.5c-3.309 0-6-2.691-6-6s2.691-6 6-6a1 1 0 110 2c-2.206 0-4 1.794-4 4s1.794 4 4 4a1 1 0 110 2z"
            fill="#545c66"
          />
          <path
            d="M14.501 51.5h1v2h-1zm0 10h1v2h-1z"
            fill="#545c66"
          />
          <path
            d="M27.5 30.001h16V32h-16z"
            fill="#ccd1d9"
          />
          <path
            d="M40.7 12h2.8c2.76 0 5 2.24 5 5s-2.24 5-5 5v8.001h-16V22c-2.76 0-5-2.24-5-5s2.24-5 5-5h2.81c1.03-1.79 2.971-3 5.19-3s4.16 1.21 5.2 3z"
            fill="#e6e9ed"
          />
          <path
            d="M42.502 32.998H28.503a1 1 0 110-2h13.999a1 1 0 110 2z"
            fill="#ccd1d9"
          />
          <path
            d="M46.499 38.999H24.505a1 1 0 110-2h21.994a1 1 0 110 2zm0 3H24.505a1 1 0 110-2h21.994a1 1 0 110 2zM41.5 45.025H29.504a1 1 0 110-2H41.5a1 1 0 110 2z"
            fill="#0092db"
          />
          <circle
            cx="54.5"
            cy="1.5"
            fill="#00aaff"
            r="1"
          />
        </svg>
        <p
          class="emotion-8 emotion-9"
        >
          Reciper
        </p>
      </div>
      <nav
        class="emotion-10 emotion-11"
      >
        <div
          class="emotion-12 emotion-13"
        >
          <a
            aria-current="page"
            class="emotion-14 emotion-15 active"
            href="/"
          >
            <svg
              class="emotion-16"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M506.14 241.86c-13.421-13.423-226.16-226.16-236-236-7.808-7.813-20.472-7.813-28.28 0-10.762 10.761-222.801 222.796-236 236-7.813 7.808-7.813 20.472 0 28.285 7.808 7.808 20.472 7.808 28.28 0l5.192-5.192V492c0 11.047 8.957 20 20 20h393.336c11.043 0 20-8.953 20-20V264.953l5.187 5.192c7.813 7.808 20.477 7.808 28.286 0 7.812-7.813 7.812-20.477 0-28.286zM314.669 472H197.336V354.668h117.332zm118 0h-78V334.668c0-11.047-8.953-20-20-20H177.336c-11.047 0-20 8.953-20 20V472h-78V224.953L256 48.285l176.668 176.668zm0 0"
                fill="currentColor"
                xmlns="http://www.w3.org/2000/svg"
              />
            </svg>
            <span
              class="emotion-17 emotion-18"
            >
              Główna
            </span>
          </a>
          <a
            class="emotion-14 emotion-15"
            href="/search"
          >
            <svg
              class="emotion-16"
              fill="currentColor"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M225.474 0C101.151 0 0 101.151 0 225.474c0 124.33 101.151 225.474 225.474 225.474 124.33 0 225.474-101.144 225.474-225.474C450.948 101.151 349.804 0 225.474 0zm0 409.323c-101.373 0-183.848-82.475-183.848-183.848S124.101 41.626 225.474 41.626s183.848 82.475 183.848 183.848-82.475 183.849-183.848 183.849z"
              />
              <path
                d="M505.902 476.472L386.574 357.144c-8.131-8.131-21.299-8.131-29.43 0-8.131 8.124-8.131 21.306 0 29.43l119.328 119.328A20.74 20.74 0 00491.187 512a20.754 20.754 0 0014.715-6.098c8.131-8.124 8.131-21.306 0-29.43z"
              />
            </svg>
            <span
              class="emotion-17 emotion-18"
            >
              Szukaj
            </span>
          </a>
          <a
            class="emotion-14 emotion-15"
            href="/recipes"
          >
            <svg
              class="emotion-16"
              fill="currentColor"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M375 422c8.285 0 15-6.715 15-15V15c0-8.285-6.715-15-15-15H45C20.187 0 0 20.188 0 45v392c0 24.813 20.188 45 45 45h75v15a14.998 14.998 0 0023.32 12.48L180 485.027l36.68 24.453A15 15 0 00240 497v-15h135c8.285 0 15-6.715 15-15s-6.715-15-15-15h-15v-30zm-15-30H90V30h270zM45 30h15v362H45c-5.258 0-10.305.914-15 2.578V45c0-8.27 6.73-15 15-15zm0 422c-8.27 0-15-6.73-15-15s6.73-15 15-15h75v30zm165 16.973l-21.68-14.453a14.998 14.998 0 00-16.64 0L150 468.973V422h60zM240 422h90v30h-90zm0 0"
                xmlns="http://www.w3.org/2000/svg"
              />
              <path
                d="M150 232.973V317c0 8.285 6.715 15 15 15h120c8.285 0 15-6.715 15-15v-84.027c18.297-10.578 30-30.313 30-51.973 0-30.719-23.203-56.121-53.008-59.594C266.477 102.867 246.722 91 225 91s-41.477 11.867-51.992 30.406C143.203 124.88 120 150.281 120 181c0 21.66 11.703 41.395 30 51.973zm29.508-81.97c.539.075 1.328.185 2.281.24a14.971 14.971 0 0015.047-10.04C201.046 129.117 212.363 121 225 121s23.953 8.117 28.164 20.203a14.999 14.999 0 0015.047 10.04c.953-.055 1.742-.165 2.281-.24C286.81 151.27 300 164.622 300 181c0 12.66-7.996 24.004-19.902 28.258-6.98 2.496-14.825 2.328-21.77-.61-7.633-3.226-16.433.344-19.66 7.973-3.227 7.633.344 16.434 7.973 19.66A59.717 59.717 0 00270 241v61h-90v-61a59.65 59.65 0 0023.36-4.723c7.628-3.226 11.199-12.027 7.972-19.656-3.227-7.629-12.027-11.2-19.66-7.973-6.934 2.934-14.781 3.11-21.77.61C157.996 205.004 150 193.66 150 181c0-16.379 13.191-29.73 29.508-29.996zm0 0"
                xmlns="http://www.w3.org/2000/svg"
              />
            </svg>
            <span
              class="emotion-17 emotion-18"
            >
              Przepisy
            </span>
          </a>
          <a
            class="emotion-14 emotion-15"
            href="/settings"
          >
            <svg
              class="emotion-16"
              fill="currentColor"
              viewBox="-12 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M244.316 362.668c-58.816 0-106.668-47.852-106.668-106.668S185.5 149.332 244.316 149.332c58.813 0 106.664 47.852 106.664 106.668S303.13 362.668 244.316 362.668zm0-181.336c-41.175 0-74.668 33.496-74.668 74.668s33.493 74.668 74.668 74.668c41.172 0 74.664-33.496 74.664-74.668s-33.492-74.668-74.664-74.668zm0 0"
              />
              <path
                d="M279.793 512h-70.957c-17.066 0-32.148-12.117-35.86-28.82l-8.98-39.938c-14.637-6.207-28.59-14.312-41.687-24.21l-38.891 12.284c-16.684 5.184-34.816-1.898-43.18-16.832l-35.367-61.16c-8.473-15.148-5.613-33.793 6.848-45.527l30.14-27.711c-.914-8.129-1.363-16.191-1.363-24.086s.45-15.957 1.344-24.086l-29.973-27.582c-12.61-11.863-15.488-30.508-7.105-45.46l35.586-61.548c8.254-14.765 26.43-21.781 43.007-16.683l38.954 12.308c13.097-9.898 27.05-18.004 41.687-24.215l9-39.976C176.688 12.118 191.77 0 208.836 0h70.957c17.066 0 32.148 12.117 35.86 28.82l8.98 39.938c14.637 6.207 28.59 14.312 41.687 24.21l38.89-12.284c16.728-5.141 34.817 1.898 43.18 16.832l35.372 61.16c8.468 15.148 5.61 33.793-6.852 45.527l-30.14 27.711c.894 8.129 1.343 16.215 1.343 24.086s-.449 15.957-1.343 24.086l29.992 27.562c.043.043.086.086.148.13a37.197 37.197 0 0 1 6.957 45.331l-35.586 61.547c-8.254 14.762-26.386 21.824-43.027 16.66l-38.957-12.308c-13.098 9.898-27.05 18.004-41.684 24.215l-9.004 39.976C311.941 499.883 296.86 512 279.793 512zM125.574 385.258c3.668 0 7.274 1.258 10.176 3.648 14.656 12.055 30.633 21.356 47.55 27.586a15.988 15.988 0 0 1 10.071 11.496l10.86 48.215c.488 2.219 2.43 3.797 4.628 3.797h70.954c2.199 0 4.14-1.578 4.609-3.754l10.879-48.258a15.988 15.988 0 0 1 10.07-11.496c16.895-6.23 32.895-15.531 47.55-27.586a15.947 15.947 0 0 1 15-2.922l46.954 14.829c2.238.703 4.629-.13 5.652-1.965l35.586-61.547c1.067-1.899.68-4.5-1.023-6.164l-36.012-33.11a15.993 15.993 0 0 1-4.992-14.144c1.406-9.45 2.113-18.836 2.113-27.903 0-9.07-.707-18.457-2.113-27.906a16.011 16.011 0 0 1 4.992-14.144l36.094-33.196c1.621-1.511 2.008-4.16.832-6.27l-35.371-61.163c-1.149-2.028-3.582-2.836-5.801-2.133l-46.89 14.805c-5.165 1.644-10.794.535-14.997-2.922-14.656-12.055-30.636-21.356-47.554-27.586a15.975 15.975 0 0 1-10.067-11.496l-10.86-48.215c-.534-2.176-2.476-3.754-4.671-3.754h-70.957c-2.195 0-4.137 1.578-4.606 3.754l-10.882 48.258a16.017 16.017 0 0 1-10.067 11.496c-16.898 6.23-32.898 15.531-47.531 27.586-4.203 3.457-9.879 4.543-14.996 2.922l-46.957-14.829c-2.195-.68-4.629.13-5.652 1.965l-35.586 61.524c-1.067 1.922-.68 4.566 1.09 6.23l35.968 33.047a15.996 15.996 0 0 1 4.992 14.145c-1.41 9.449-2.113 18.836-2.113 27.902s.703 18.453 2.113 27.902c.79 5.25-1.09 10.54-4.992 14.145L33.52 331.242c-1.622 1.516-2.004 4.16-.833 6.274l35.372 61.16c1.128 2.027 3.562 2.816 5.8 2.137l46.895-14.81c1.578-.51 3.2-.745 4.82-.745zm0 0"
              />
            </svg>
            <span
              class="emotion-17 emotion-18"
            >
              Ustawienia
            </span>
          </a>
        </div>
      </nav>
    </aside>
  </div>
</DocumentFragment>
`;
